typing/includeclass.ml

```diff
@@ -70,23 +70,35 @@ let include_err ppf =
           "is not matched by the class type"
           Printtyp.class_type cty2)
   | CM_Parameter_mismatch (env, trace) ->
-      Printtyp.report_unification_error ppf env trace
+      Printtyp.report_error Printtyp.moregen ppf env trace
         (function ppf ->
           fprintf ppf "A parameter has type")
         (function ppf ->
           fprintf ppf "but is expected to have type")
   | CM_Val_type_mismatch (lab, env, trace) ->
-      Printtyp.report_unification_error ppf env trace
+      Printtyp.report_error Printtyp.moregen ppf env trace
         (function ppf ->
           fprintf ppf "The instance variable %s@ has type" lab)
         (function ppf ->
           fprintf ppf "but is expected to have type")
+  | CM_Val_type_mismatch_eq (lab, env, trace) ->
```

Comment:
I think that we can probably get away without a new constructor for this now: just a boolean or two case variant type saying whether the error was from `moregen` or `eqtype` will suffice.